lemur/migrations/versions/614f0f51b553_.py

```diff
@@ -0,0 +1,91 @@
+"""This Database upgrade will make the domains table unique and is required for future versions of Lemur.
+
+Revision ID: 614f0f51b553
+Revises: 4fe230f7a26e
+Create Date: 2021-06-22 23:22:30.078483
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '614f0f51b553'
+down_revision = '4fe230f7a26e'
+
+import datetime
+import time
+from logging import Formatter, FileHandler, getLogger
+
+from alembic import op
+from flask import current_app
+from sqlalchemy.sql import text
+
+log = getLogger(__name__)
+handler = FileHandler(current_app.config.get("LOG_UPGRADE_FILE", "db_upgrade.log"))
+handler.setFormatter(
+    Formatter(
+        "%(asctime)s %(levelname)s: %(message)s " "[in %(pathname)s:%(lineno)d]"
+    )
+)
+handler.setLevel(current_app.config.get("LOG_LEVEL", "DEBUG"))
+log.setLevel(current_app.config.get("LOG_LEVEL", "DEBUG"))
+log.addHandler(handler)
+
+
+def upgrade():
+    log.info("\n*** Starting DB upgrade(%s) ***\n" % datetime.datetime.now())
+    start_time = time.time()
+
+    # Do Upgrade
+    dedup_domains_table()
+
+    log.info("---Upgrade Complete - Total %s seconds ---\n" % (time.time() - start_time))
+
+
+def downgrade():
+    log.info("\n*** Request to downgrade at (%s) ***\n" % datetime.datetime.now())
+    log.info("It is not possible to downgrade this change since duplicated domains were already deleted.")
+    log.info("If you need to restore the duplicated domains for some reason, you will need to restore from backup.")
+
+
+def dedup_domains_table():
+    conn = op.get_bind()
+
+    # Loop through all domains and get a list of them by count
+    for name, count in conn.execute(text(
+            "SELECT name,count(*) FROM domains GROUP BY name HAVING COUNT(*) > 1 ORDER BY count(*) ASC")):
```

Comment:
Not sure if this buys much improvement, but how about removing `ORDER BY count(*) ASC`. I could've missed it, the script does not use sorting specifically.

Reply:
That is correct.  It is done this way to start with the domains with less duplicates which made debugging the code easier.  There is no tangible reason to leave it in now, but the the improvement would be negligible.